cmd/404-server-with-metrics/server-with-metrics.go

```
@@ -245,6 +250,25 @@ func (s *server) notFoundHandler() http.HandlerFunc {
 	}
 }
 
+// statusCodeHandler returns a status code given by URL path in the request.
+// If URL path does not contain a valid status code, returns 404 (NotFound).
+func (s *server) statusCodeHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		p := strings.TrimPrefix(r.URL.Path, statusCodePrefix)
+		code, err := strconv.Atoi(p)
+		if err != nil {
+			code = http.StatusNotFound
```

we should return an error message here instead of dropping through...

---

Done.